clang/lib/AST/MicrosoftMangle.cpp

```diff
@@ -2808,6 +2812,7 @@ void MicrosoftCXXNameMangler::mangleType(const BuiltinType *T, Qualifiers,
 
 #define SVE_TYPE(Name, Id, SingletonId) \
   case BuiltinType::Id:
+#define SVE_SCALAR_TYPE(Name, MangledName, Id, SingletonId, Bits)
```

Comment:
Is there a reason why this can't be handled by defining `SVE_SCALAR_TYPE` to the the mangling? 
I was a little confused about what this empty define was doing at first.

```suggestion
#define SVE_SCALAR_TYPE(Name, MangledName, Id, SingletonId, Bits)  \
    mangleArtificialTagType(TagTypeKind::Struct, MangledName);     \
    break;
```
Also, is there anyone we can tag that knows about the Microsoft mangling? I'm not sure if there's any process needed for adding a new type.

Reply:
It can be done. Doesn't mean it's a good idea.
With the proposed change all the SVE types will fall-through to the `SVE_SCALAR_TYPE` case, or worse
some will and some won't, depending on the relative order in `AArch64SVEACLETypes.def`.
That can work by including the `.def` file twice, once for the scalar type and once for all the SVE types.
Or use a `default:` for the unsupported cases, but I decided to stick to the existing style.

Reply:
> Also, is there anyone we can tag that knows about the Microsoft mangling? I'm not sure if there's any process needed for adding a new type.

No idea. I just made sure the results are consistent with mangling of `__bf16` , `int8_t`, and vectors of those.

Reply:
With the current approach there simply won't be cases added for other `SVE_SCALAR_TYPE` types, so they won't be covered by the switch statement at all (which does not appear to have a `default:`). There will probably a warning when building clang, but it looks like at runtime the compiler will just silently drop mangling these types (and won't error).

Reply:
You could define the macro to check the builtin-type is the one you expect (I think `BuiltinType::MFloat8` is the only SVE_SCALAR_TYPE right now). 
```
#define SVE_SCALAR_TYPE(Name, MangledName, Id, SingletonId, Bits)  \
    case BuiltinType::Id:                                          \
      if (BuiltinType::Id == BuiltinType::MFloat8) {                  \
        mangleArtificialTagType(TagTypeKind::Struct, MangledName);   \
        break;                                                                                            \
      }                                                                                                         \
      [[fallthrough]];
```

Reply:
> With the current approach there simply won't be cases added for other `SVE_SCALAR_TYPE` types.
Good point. Addressed in https://github.com/llvm/llvm-project/pull/125051